### PR TITLE
Document source code folder structure

### PR DESCRIPTION
**Pull request by hobyst**
# Overview
Documents folder structure of source code files related to the emulator to make it easier for other people to contribute to the project.

# Changes
- **Moved screenshot files used on `README.md` to `./_readme`:** Done in order to reserve `./docs` for a future and proper user/developer manual. It is a common practice to keep project documentation side by side to the source code to keep things unified and it would also allow to use proper documentation tools such as Sphinx, which are much more capable than GitHub wikis.
- **Added a Doxyfile:** For easier documentation access. The file will generate a Doxygen HTML output on `./docs/html` with the project's icon and description. Documentation can be generated by running:
  ```bash
  cd docs
  doxygen Doxyfile
  ```
- **Added descriptions for source code folders:** Using `.cpp` files made out of Doxygen formatted comments, these will allow Doxygen to show descriptions to each source code folder on the file list view for easier file navigation.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,5 @@ cmake-build-*/
 build
 # generated on macOS
 .DS_Store
+# generated by Doxygen
+/docs/html/
```
Comment (Zangetsu38) [inline .gitignore:15]: add new line :)

File: docs/Doxyfile [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+DIAFILE_DIRS           =
+PLANTUML_JAR_PATH      =
+PLANTUML_CFG_FILE      =
+PLANTUML_INCLUDE_PATH  =
+DOT_GRAPH_MAX_NODES    = 50
+MAX_DOT_GRAPH_DEPTH    = 0
+DOT_TRANSPARENT        = NO
+DOT_MULTI_TARGETS      = NO
+GENERATE_LEGEND        = YES
+DOT_CLEANUP            = YES
```
Comment (Zangetsu38) [inline docs/Doxyfile:392]: add new line

File: tools/dir_doc.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+* Needs VitaSDK to be compiled
+*/
+
+/**
+* @dir tools/usse-decoder-gen
+* 
+* @brief Instruction bitmask generator for the PS Vita GPU's Universal Scalable Shader Engine instruction set
+* 
+* Needs VitaSDK to be compiled
+*/
```
Comment (Zangetsu38) [inline tools/dir_doc.cpp:38]: same

File: vita3k/dir_doc.cpp [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+* @dir vita3k/touch
+* 
+* @brief PS Vita touchscreen emulation
+*/
+
+/**
+* @dir vita3k/util
+* 
+* @brief Several code utils used throughout the source code
+*/
```
Comment (Zangetsu38) [inline vita3k/dir_doc.cpp:217]: same

**Comment by Zangetsu38** [comment]
no create nes commiçt for fix any things, edit commit, and here you can squash all in one, no need fews commit for that

**Comment by VelocityRa** [comment]
The commits look fine to me.

However, I'm not sure about the Doxygen additions.
I don't think I have ever seen a Doxygen doc that was ever worth reading. Especially for open-source projects. The code is the documentation (but unironically).
I appreciate the effort but this project is in need of core contributors, not docs. And no active dev is going to bother writing docs when there's so much other stuff to do as far as development goes.

There are many more reasons, but that's the gist of it.

It would be better if instead of Doxygen, your docs were were just in a `CONTRIBUTING.md` file or something, like for some other projects. That could be useful.

**Comment by Zangetsu38** [comment]
maybe, but need remove last commit, is just fix review, no have sense keep it, need set this change insiçde intialcommit

**Review by Zangetsu38 — APPROVED** [review]

**Comment by VelocityRa** [comment]
Well thanks for ignoring the rest of my comment & merging anyway.

**Comment by Zangetsu38** [comment]
oh sorry :/

**Comment by hobyst** [comment]
> However, I'm not sure about the Doxygen additions.
> I don't think I have ever seen a Doxygen doc that was ever worth reading. Especially for open-source projects. The code is the documentation (but unironically).

That's the issue. Sometimes is way more easier to look and understand how a software works by looking at comments than trying to read the code line by line. Doxygen, in the end, generates its documentation from parsing comments, so if a Doxygen doc is not worth reading is usually due to either the project not being properly commented or the devs didn't bother to create formatted comment blocks.

> I appreciate the effort but this project is in need of core contributors, not docs. And no active dev is going to bother writing docs when there's so much other stuff to do as far as development goes.

I'm mainly doing this because I want to start to contribute to the project at some point, but first I need to know how it is organized and structured internally so I know exactly where to do what when I need to. I could take annotations on a separate document or something, but doing this makes more sense to me as it would be reusable by any other person who wanted to contribute to the project in the future at some point (if there are any).

Rather than just writing comments for everything, I will document code as I modify stuff on the project, so the code base I end up dealing with will get more or less documented. Not all the project will end up getting documented, but at least some of it.

> It would be better if instead of Doxygen, your source tree descriptions were were just in a `CONTRIBUTING.md` file or something, like for some other projects. That _could_ be useful & not much effort to maintain.

While it would be easier to just do it on a `CONTRIBUTING.md` file, if the code base starts to get more commented and documented through formatted comments, why would the folder structure be stored elsewhere in the project when it could just be displayed along with the rest of code documentation on the Doxygen docs?

**Pull request merged**